File: Largest Number/Largest_Number.java
class Solution {
    public String largestNumber(int[] nums) {
        List<String> numStrs = new ArrayList<>();
        for (int num : nums) {
            numStrs.add(Integer.toString(num));
        }
        
        Collections.sort(numStrs, new Comparator<String>() {
            public int compare(String a, String b) {
                String ab = a + b;
                String ba = b + a;
                return ba.compareTo(ab);
            }
        });
        
        StringBuilder sb = new StringBuilder();
        for (String s : numStrs) {
            sb.append(s);
        }
        String ans = sb.toString();
        
        if (ans.charAt(0) == '0') {
            return "0";
        }
        return ans;
    }
}

File: Largest Number/Largest_Number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> arr;
        for(auto i: nums) {
            string x = to_string(i);
            arr.push_back(x);
        }
        sort(arr.begin(),arr.end(),[&](string a,string b) {
            // true, false;
            // true -> a b
            // false -> b a
            string ab = a+b;
            string ba = b+a;
            if(ab > ba) return true;
            else return false;
        });
        string ans = "";
        for(auto i: arr) {
            ans += i;
        }
        if(ans[0] == '0') return "0";
        return ans;
    }
};